zenodo/modules/stats/exporters.py

```diff
@@ -0,0 +1,135 @@
+            time_range['gte'] = start_date.replace(microsecond=0).isoformat()
+        if end_date is not None:
+            time_range['lte'] = end_date.replace(microsecond=0).isoformat()
+
+        events = Search(
+            using=current_search_client,
+            index='events-stats-*'
+        ).filter(
+            'range', timestamp=time_range
+        ).sort(
+            {'timestamp': {'order': 'asc'}}
+        ).params(preserve_order=True).scan()
+
+        url = current_app.config['ZENODO_STATS_PIWIK_EXPORTER'].get('url', None)
+        token_auth = current_app.config['ZENODO_STATS_PIWIK_EXPORTER'] \
+            .get('token_auth', None)
+
+        chunk_size = 85
+        for event_chunk in chunkify(events, chunk_size):
+            query_strings = []
+            for event in event_chunk:
+                query_string = self._build_query_string(event)
+                query_strings.append(query_string)
+
+            payload = {
+                'requests': query_strings,
+                'token_auth': token_auth
+            }
+
+            res = requests.post(url, json=payload)
+
+            # Failure: not 200 or not "success"
+            content = json.loads(res.content)
+            if res.status_code == 200 and content.get('status') == 'success':
+                if content.get('invalid') != 0:
+                    msg = 'Invalid events in Piwik export.'
+                    extra = {
+                        'begin_event': event_chunk[0].timestamp,
+                        'end_event': event_chunk[len(event_chunk)-1].timestamp,
+                        'invalid_events': content.get('invalid')
+                    }
+                    current_app.logger.warning(msg, extra=extra)
+                elif update_bookmark is True:
+                    current_cache.set('piwik_export:bookmark',
+                                      event_chunk[len(event_chunk)-1].timestamp,
+                                      timeout=-1)
+            else:
+                msg = 'Invalid events in Piwik export.'
+                extra = {
+                    'begin_event': event_chunk[0].timestamp,
+                    'end_event': event_chunk[len(event_chunk) - 1].timestamp
+                }
+                current_app.logger.warning(msg, extra=extra)
+                raise Exception()
+
+    def _build_query_string(self, event):
+        _, record = fetch_record(event.recid)
+        id_site = current_app.config['ZENODO_STATS_PIWIK_EXPORTER']\
+            .get('id_site', None)
+        url = 'zenodo.org/record/{}'.format(event.recid)
+        visitor_id = event.visitor_id[0:16]
+        oai = record.get('_oai').get('id')
+        cvar = "{\"1\":[\"oaipmhID\",\"" + oai + "\"]}"
+        action_name = record.get('title')[:150]  # max 150 characters
+
+        if 'file_key' in event:
+            url = '{}/files/{}'.format(url, event.file_key)
+            query_string = '?idsite={}&rec=1&url={}&_id={}&cid={}&cvar={}&' \
```

Comment:
To keep it safe in terms of URL-encoding, you can use [urllib.parse.urlencode](https://docs.python.org/3/library/urllib.request.html#urllib-examples) and also simplify the file-specific logic by using a dictionary with the values. Unfortunately, since we're targeting both Python 2 + 3 you have to use `from six.moves.urllib.parse import urlencode` :sob: